src/indexing.jl

```diff
@@ -0,0 +1,231 @@
+
+_maybe_tail(::Tuple{}) = ()
+_maybe_tail(x::Tuple) = tail(x)
+
+#
+#    to_indices(A, args)
+#
+function to_indices end
+const to_indices = ArrayInterface.to_indices
```

Comment:
Redundant?

Reply:
So that all `to_indices` methods in the module go to the one specified here instead of the one in base.

Reply:
Isn't the `function to_indices end` enough?
```julia
julia> function inv end
inv (generic function with 0 methods)

julia> inv(x) = 2x
inv (generic function with 1 method)

julia> Base.inv(3)
0.3333333333333333

julia> inv(3)
6
```

Reply:
Oh, you're totally right. I was basing this off of what happens when exporting it into a new module where `const to_indices = ArrayInterface.to_indices` was necessary.